File: Challenges/challenge_01.java
// The challenge today is very easy we want you to convert this code from java to cpp :)

import java.util.Scanner;

class Calculator{

    public double summation(double num1,double num2){
        
        return num1 + num2;
    }
    
    public double subtraction(double num1,double num2){
 
        return num1 - num2;
    }
    
    public double multiplication(double num1,double num2){
 
        return num1 * num2;
    }
    
    public double divison(double num1,double num2){
 
        return num1 / num2;
    }


public static void main(String[] args) {
Scanner in=new Scanner(System.in);
Calculator result=new Calculator();
char op;
double num1,num2;
System.out.print("Enter first number : ");
num1=in.nextDouble();
System.out.print("Enter operator (+,-,*,/) : ");
op=in.next().charAt(0);
System.out.print("Enter second number : ");
num2=in.nextDouble();
switch (op) {
    case '+':
        System.out.println(num1+" + "+num2+" = "+result.summation(num1, num2));
        break;
    case '-':
        System.out.println(num1+" - "+num2+" = "+result.subtraction(num1, num2));
        break;
    case '*':
        System.out.println(num1+" * "+num2+" = "+result.multiplication(num1, num2));
        break;
    case '/':
        System.out.println(num1+" / "+num2+" = "+result.divison(num1, num2));
        break;
    default:
        System.out.println("Invalid operator");
        break;
}
}
}

File: Challenges/challenge_01.cpp
#include <iostream>

using namespace std;

class Calculator
{

public:
    double summation(double num1, double num2)
    {
        return num1 + num2;
    }

    double subtraction(double num1, double num2)
    {
        return num1 - num2;
    }

    double multiplication(double num1, double num2)
    {
        return num1 * num2;
    }

    double division(double num1, double num2)
    {
        return num1 / num2;
    }
};

int main()
{
    Calculator result;
    char op;
    double num1, num2;

    cout << "Enter first number: ";
    cin >> num1;

    cout << "Enter operator (+, -, *, /): ";
    cin >> op;

    cout << "Enter second number: ";
    cin >> num2;

    switch (op)
    {
    case '+':
        cout << num1 << " + " << num2 << " = " << result.summation(num1, num2) << endl;
        break;
    case '-':
        cout << num1 << " - " << num2 << " = " << result.subtraction(num1, num2) << endl;
        break;
    case '*':
        cout << num1 << " * " << num2 << " = " << result.multiplication(num1, num2) << endl;
        break;
    case '/':
        if (num2 != 0)
            cout << num1 << " / " << num2 << " = " << result.division(num1, num2) << endl;
        else
            cout << "Infinity" << endl;
        break;
    default:
        cout << "Invalid operator" << endl;
        break;
    }

    return 0;
}
